v3.6.x Bugfixes - See issue list and target this branch for related PRs

zeel01 opened the pull request:
This is a tracking PR for various high-priority issues:

Planned fixes:
- #197
- #206 
- #215 
- #231
- #232 
- #233 
- #230 / #241 
- #242
- #243

Unplanned (but maybe):
- #65

zeel01 committed:
v3.6.x branch

Aioros commented:
Here's a quick rundown from some local testing in my main three instances (v11/v3, v12/v3, v12/v4). Tried

Probably stale:
- #197: could not repro;
- #206: could not repro;
- #231: could not repro;
- #233: could not repro (should be already fixed);

Partially stale:
- #215: could not repro in v12/v3 or v12/v4. In v11/v3, setting a condition immunity caused it to be displayed as `[object set]` and then unable to modify;

Current:
- #232: verified current issue;
- #241: verified current issue.

I didn't dive into the commit history yet to check which ones of these "could not repro" have actually been specifically targeted and fixed already, but I thought this would give at least some more insight into the current status and/or prioritization. I'll add info when I have more.

zeel01 commented:
Much appreciated! I'll comment on the ones you couldn't reproduce and ask the reporters if they are still having the problem in the new version.

zeel01 commented:
I feel like #233 wouldn't have fixed itself, I even have a comment in the code there because I knew it was going to be a problem. But there's a good chance one of you guys did fix it because it should have been a big problem.

#206... Input Expressions... right...

#215 if this is resolved for v12 but not for v11 I'm okay with that.

Aioros commented:
Fairly sure #233 was addressed by @erizocosmico in their v12/v3 branch (that I included).

zeel01 commented:
#215 and #233 are checked off.

zeel01 commented:
Added #242

Aioros committed:
- Fixed description enrichment (#241);
- Fixed Legendary Actions section title (#242);
- Fixed old reference to .data and additional deprecations (#243);
- Fixed reference to ability.save (now .save.value in 4.x) (#245);
- Fixed issue with inventory items (#247).
- Improved innate spellcasting management with initial support for cast activities (like in the MM2024) (#246).

zeel01 committed:
Merge pull request #248 from Aioros/v3.6.x

Misc fixes to the most recent issues

Aioros committed:
Fixed displaying custom values for resistances and immunities (#232)

Aioros commented:
While I look at a couple other issues, question about Input Expressions and math.js. There's a simple [PR](https://github.com/zeel01/input-expressions/pull/11) on IE's repo that would solve our issues, but I'm also curious if you still want to keep both the git submodule and the math.js Foundry module dependency. Since IE already includes the math.js script in its folder, we could load that one directly and remove the dependency. Let me know if that sounds reasonable.

Aioros committed:
Merge pull request #249 from Aioros/v3.6.x

Fixed displaying custom values for resistances and immunities (#232)

Aioros commented:
Yeah, it's a bit tricky. But also, it's fair to say that it's not something that actually needs to be changed, so we're good (I do recommend updating the Math.js manifest, at least, to avoid the annoying warnings in the setup screen).

If you're ok with it, I think this branch is in a good spot for a new release, unless you want us to look into that #65 first. I personally suggest postponing it just because the module is currently not working on dnd5e 4.3.x, and I've seen a few users anxious for an update already. We could push this one out to get in a stable place, and maybe close up a few old issues before going for enhancements.

By the way, never worked with git submodules. I suppose IE it needs to be updated here somehow to point to the new commit?

zeel01 commented:
Yes, we need to update the submodule too, I don't even remember how that's done. https://git-scm.com/book/en/v2/Git-Tools-Submodules naturally, it's overcomplicated.

Aioros commented:
Lol, ok, let me take a look.

Aioros committed:
Updated input-expression submodule, removed obsolete manifest keys

zeel01 committed:
Merge pull request #250 from Aioros/v3.6.x

Updated input-expression submodule, removed obsolete manifest keys
